lib/cli/src/commands/run/mod.rs

```diff
@@ -140,9 +144,75 @@ impl Run {
         #[cfg(not(feature = "sys"))]
         let engine = store.engine().clone();
 
-        let runtime =
-            self.wasi
-                .prepare_runtime(engine, &self.env, runtime, preferred_webc_version)?;
+        let registry_name = self
+            .env
+            .registry_public_url()?
+            .host_str()
+            .unwrap_or("unknown_registry")
+            .replace('.', "_");
+
+        // We don't have the bytes of the module yet, but we still want to have the
+        // package-capabilities cache be as close to an actual identifier as possible.
+        let package_cache_path = match &self.input {
+            PackageSource::File(f) => {
+                let full_path = f.canonicalize()?.to_path_buf();
+                let mut h = DefaultHasher::new();
```

Comment:
This hasher might not be consistent over rust versions. We need to use a consistent one

Reply:
I'd probably recommend using something like xxhash or other fast hasher

Reply:
Or a cryptographically secure hasher (like blake3 or sha)

Reply:
[9f9c47a](https://github.com/wasmerio/wasmer/pull/4975/commits/9f9c47a2f75643b1eca6ce8a22ba3ead2d63553f)